### ISSUE TITLE
Prefer scoped enums where possible

### ISSUE THREAD

**Issue by ephphatha** [diasurgical/devilutionX#2300]
Many of the unscoped enums used in devilutionX can use `enum class` instead. This gives stronger type safety for things like `Direction` and allows unambiguous constructors/conversion functions in related classes (e.g. `Displacement`).

This does make it more difficult to cast back to an integral type so the usage of some enums (such as `_item_indexes`) will need to be evaluated to see if it's worth changing those from unscoped to scoped.

**Comment by AJenbo** [comment]
Approved

**Comment by ephphatha** [comment]
@AJenbo While looking at UI related code I've noticed some enums were converted from scoped to unscoped to enable compiling in C++03:
https://github.com/diasurgical/devilutionX/commit/8726e80199dcb1af10206aacc27ccec278d8836a - Enums related to controller buttons

Couldn't find a PR for that commit but I did find the following as part of C++03 compatibility/xbox support:
https://github.com/diasurgical/devilutionX/pull/736 and https://github.com/diasurgical/devilutionX/pull/731 - Devilution specific versions of windows constants? Mostly defines (previously constexpr values) but these could be grouped into enums for keycodes and windows messages.

Is supporting C++03 still required? If not reverting 8726e80 gets rid of a lot of warnings pretty quickly :D

**Comment by obligaron** [comment]
I think this is obsolete since #1570 (use c++ 17) got merged. 🙂 
So we can get rid of the unscoped enums.  👍

**Comment by AJenbo** [comment]
Yeah a lot of the duming down of the code should in practice be reverted.

Source/miniwin/misc.h should go away, probably they should be redefined as events and not key board keys. Can be tied in with the keymapper etc
